tests/rbd/test_rbd_trash_purge_schedule.py

```diff
@@ -0,0 +1,159 @@
+import random
+import time
+
+from ceph.parallel import parallel
+from tests.rbd.rbd_utils import Rbd, initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def create_an_image_run_fio(rbd, pool, client, image_name):
+    if rbd.create_image(pool, image_name, "10G"):
+        log.error(f"Image creation failed for {image_name}")
+        return 1
+    run_fio(image_name=image_name, pool_name=pool, client_node=client, runtime=10)
+    return 0
+
+
+def create_images_run_fio_in_parallel(rbd, pool, client, num_images):
+    log.info(f"Creating {num_images} images in pool {pool} and initiating io")
+
+    with parallel() as p:
+        for i in range(0, num_images):
+            p.spawn(create_an_image_run_fio, rbd, pool, client, f"image_{i}")
+        return 0
+
+
+def move_all_images_to_trash(rbd, pool, num_images):
+    log.info("Moving all images in pool to trash")
+    for i in range(0, num_images):
+        rbd.move_image_trash(pool, f"image_{i}")
+
+
+def prepare_images_and_move_to_trash(rbd, pool, client, num_images):
+    create_images_run_fio_in_parallel(rbd, pool, client, num_images)
+    move_all_images_to_trash(rbd, pool, num_images)
+
+
+def verify_schedule_add(
+    rbd: Rbd,
+    verification_iterations: int,
+    schedule_interval: int,
+    pool: str,
+    client,
+    num_images: int,
+) -> int:
+    iterations_remaining = verification_iterations
+    log.debug(f"Verifying purge on pool {pool}")
+    while iterations_remaining > 0:
```

Comment:
looks Iteration is not matching with the count.
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-09IEHV/Test_to_verify_trash_purge_schedule_0.log
above log shows "Trash purge schedule will be tested for an interval of 2 minutes for 5 iterations"
but it running 6 times for 2 min interval.
I think we need to check condition.

Reply:
Issue resolved